tjreports/site/controllers/reports.php

```diff
@@ -350,4 +352,45 @@ public function getreport()
 			jexit();
 		}
 	}
+
+/**
+	* Function used to delete reports
+	*
+	* @return  boolean
+	*
+	* @since  1.0
+	*/
+	public function deleteQuery()
+	{
+		$cid = JFactory::getApplication()->input->get('cid', '', 'array');
+		$model = JModelLegacy::getInstance('Report', 'TjreportsModel');
+
+		if ($model->delete($cid))
+		{
+			echo json_encode(1);
+			jexit();
+		}
+		else
+		{
+			return false;
+		}
+	}
+
+// This is useful for manager report
+/**
+	* Function used to delete reports
+	*
+	* @return  json
+	*
+	* @since  1.0
+	*/
+	public function setUserType()
+	{
+		$userTypeId = JFactory::getApplication()->input->get('userTypeId', '', 'int');
+
+		$setUserType = JFactory::getApplication()->setUserState("setUserType", $userTypeId);
```

Comment:
@hemantpurohit1  I think while setting variable in userstate, you should also consider context.